internal/operands/common-templates/reconcile.go

```diff
@@ -211,16 +209,13 @@ func reconcileOlderTemplates(request *common.Request) ([]common.ReconcileFunc, e
 			continue
 		}
 
-		if template.Annotations == nil {
-			template.Annotations = make(map[string]string)
-		}
-		template.Annotations[TemplateDeprecatedAnnotation] = "true"
 		funcs = append(funcs, func(*common.Request) (common.ResourceStatus, error) {
 			return common.CreateOrUpdate(request).
 				ClusterResource(template).
 				WithAppLabels(operandName, operandComponent).
 				UpdateFunc(func(_, foundRes client.Object) {
 					foundTemplate := foundRes.(*templatev1.Template)
+					foundTemplate.Annotations[TemplateDeprecatedAnnotation] = "true"
```

Comment:
Will this be safe enough? Are we sure that our older templates always have annotations already?

Reply:
This function will tag as deprecated all templates which have different version, then is current version. The old templates don't have to have this label.

Reply:
I think Kevin means, that this could panic, if `foundTemplate.Annotations == nil`. In normal situation this should not happen, but user could remove them.

Reply:
if user removes annotations, it will not panic, because before operator calls that `UpdateFunc` (which adds the deprecated annotation), operator framework creates ownerAnnotations (https://github.com/kubevirt/ssp-operator/blob/master/internal/common/resource.go#L123), which creates new `map[string]string{}` (https://github.com/kubevirt/ssp-operator/blob/master/vendor/github.com/operator-framework/operator-lib/handler/enqueue_annotation.go#L170)

Reply:
Ok, I missed that.